typings/algoliasearch.d.ts

```diff
@@ -0,0 +1,1069 @@
+     * 'prefixAll' All query words are interpreted as prefixes. This option is not recommended.
+     * 'prefixLast' Only the last word is interpreted as a prefix (default behavior).
+     * 'prefixNone' No query word is interpreted as a prefix. This option is not recommended.
+     */
+    queryType?: any;
+    /**
+     * This option is used to select a strategy in order to avoid having an empty result page
+     * default: 'none'
+     * 'lastWords' When a query does not return any results, the last word will be added as optional
+     * 'firstWords' When a query does not return any results, the first word will be added as optional
+     * 'allOptional' When a query does not return any results, a second trial will be made with all words as optional
+     * 'none' No specific processing is done when a query does not return any results
+     */
+    removeWordsIfNoResults?: string;
+    /**
+     * Enables the advanced query syntax
+     * default: false
+     */
+    advancedSyntax?: boolean;
+    /**
+     * A string that contains the comma separated list of words that should be considered as optional when found in the query
+     * default: []
+     */
+    optionalWords?: string[];
+    /**
+     * Remove stop words from the query before executing it
+     * default: false
+     * true|false: enable or disable stop words for all 41 supported languages; or
+     * a list of language ISO codes (as a comma-separated string) for which stop words should be enable
+     */
+    removeStopWords?: string[];
+    /**
+     * List of attributes on which you want to disable prefix matching
+     * default: []
+     */
+    disablePrefixOnAttributes?: string[];
+    /**
+     * List of attributes on which you want to disable the computation of exact criteria
+     * default: []
+     */
+    disableExactOnAttributes?: string[];
+    /**
+     * This parameter control how the exact ranking criterion is computed when the query contains one word
+     * default: attribute
+     * 'none': no exact on single word query
+     * 'word': exact set to 1 if the query word is found in the record
+     * 'attribute': exact set to 1 if there is an attribute containing a string equals to the query
+     */
+    exactOnSingleWordQuery?: string;
+    /**
+     * Specify the list of approximation that should be considered as an exact match in the ranking formula
+     * default: ['ignorePlurals', 'singleWordSynonym']
+     * 'ignorePlurals': alternative words added by the ignorePlurals feature
+     * 'singleWordSynonym': single-word synonym (For example "NY" = "NYC")
+     * 'multiWordsSynonym': multiple-words synonym
+     */
+    alternativesAsExact?: any;
+    /**
+     * The name of the attribute used for the Distinct feature
+     * default: null
+     */
+    attributeForDistinct?: string;
+    /**
+     * If set to 1, enables the distinct feature, disabled by default, if the attributeForDistinct index setting is set.
+     */
+    distinct?: number;
```

Comment:
yes yes, for dynamically typed param, we ll use any !
thanks for spotting it out !